encode.lisp

```diff
@@ -206,13 +206,15 @@
   `(let ((*json-output* (make-instance 'json-output-stream :output-stream ,stream ,@args)))
      ,@body))
 
-(defmacro with-output-to-string* ((&rest args &key indent) &body body)
+(defmacro with-output-to-string* ((&rest args &key indent stream-symbol) &body body)
   "Set up a JSON streaming encoder context, then evaluate BODY.
-Return a string with the generated JSON output."
+  Return a string with the generated JSON output."
   (declare (ignore indent))
-  `(with-output-to-string (s)
-     (with-output (s ,@args)
-       ,@body)))
+  (let ((stream (or stream-symbol (gensym "STREAM"))))
```

Comment:
Is there a specific reason why you did not declare the `stream-symbol` argument as `(stream-symbol (gensym "STREAM"))` instead?

Reply:
Yes. Outer-level macro writing is easier, because this argument can always be specified.

Basically, you don't need to do
```lisp
   (yason:with-output-to-stream* (:indent ,indent-arg
                                  ,@ (if stream-arg
                                        `(:stream-symbol ,stream-arg)))
      ...)
```

but can just do
```lisp
   (yason:with-output-to-stream* (:indent ,indent-arg
                                  :stream-symbol ,stream-arg)
      ...)
```
and the macro does the right thing.

Reply:
Well, "the right thing" may vary, but as I'm not writing a lot of CL these days, I can live with this.

Reply:
So I should change that to a default value for the argument instead?
Or "can you live" with my `OR` clause?